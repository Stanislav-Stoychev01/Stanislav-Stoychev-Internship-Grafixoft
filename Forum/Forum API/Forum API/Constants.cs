namespace Forum_API
{
    public static class Constants
    {
        public const string defaultRoleDescription = "default user";
        public const string dateFormat = "yyyy - MM - dd HH:mm:ss";
    }
}
