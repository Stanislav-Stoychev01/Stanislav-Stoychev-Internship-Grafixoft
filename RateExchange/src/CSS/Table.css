#table-component {
    display: flex;
    justify-content: center;
    padding: 2%;
    border-collapse: collapse;
}

.table-head > tr > th {
    padding: 2%;
    background-color: #14631496;
    border: 2pt solid black;
}

.table-body {
    display: flex;
    justify-content: center;
}

.table-row, .table-head, .table-col {
    background-color: rgb(136, 224, 140);
    border: 2pt solid black;
}

.table-col {
    text-align: center;
}

.current-date {
    display: flex;
    justify-content: center;
    padding-top: 2%;
}

.exchange-date-label {
    font-size: 200%;
}
