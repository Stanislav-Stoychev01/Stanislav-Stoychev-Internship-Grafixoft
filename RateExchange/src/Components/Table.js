import { useEffect, useState } from 'react';
import { isElement } from 'react-dom/test-utils';
import { FetchCurrency } from '../Utils/FetchAPI';
import * as Utils from '../Utils/FilterResponse';

function Table(URL)
{
    const [items, setItems] = useState([]);

    useEffect(() => {
        const currencyChangeRate = async () => {
            const data = await FetchCurrency('cad');
            const currenctyList = Utils.filterCurrencies(data);
            const sortIntoArrays = Utils.ArrayGroups(currenctyList); 
            const convertToArray = Object.values(sortIntoArrays);    
            console.log(convertToArray);
            setItems(convertToArray);
        }

        currencyChangeRate()
        .catch((err) => console.log(err));

    }, []);
    
    return (    
        <div>
        <table id="table">
          <thead>
            <tr>
              <th>Conversion Rates below 1</th>
              <th>Conversion Rates between 1 and 1,5</th>
              <th>Conversion Rates above 1,5</th>
            </tr>
          </thead>
            <tbody id="table-body">
            <tr>
            {
                items.map((array, arrayIndex) => {
                    return (
                        <td key = { arrayIndex }>
                            <table>
                                <tbody>
                                    {
                                        array.map((item, itemIndex) => {
                                            return (
                                                <tr key = {itemIndex}>
                                                    <td>{item[0]} : {item[1]}</td>
                                                </tr>
                                            )
                                        })
                                    }
                                </tbody>
                            </table>
                            {
                                array.length > 0 &&
                            <table>
                                <tbody>
                                    <tr>
                                        <td>
                                            { array.length }
                                        </td>
                                    </tr>
                                </tbody>
                            </table>
                            }
                        </td>             
                )})
            }
            </tr>
            </tbody>
          </table>
        </div>
    );
}

export default Table;
