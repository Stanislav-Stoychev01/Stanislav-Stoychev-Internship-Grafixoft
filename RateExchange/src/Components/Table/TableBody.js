
import * as CurrencyFilters from '../../Utils/CurrencyFilters';


function TableBody(props) {

    const TableData = () => {
        return (
            CurrencyFilters.convertToRows(props.data).map((array, arrayIndex) => {
                return (
                    <tr key={arrayIndex} className="table-row">
                        {
                            array.map((item, itemIndex) => {
                                const undefinedItem = [undefined, undefined];
                                const [firstItem, secondItem] = item ?? undefinedItem;
                                return (
                                    <td key={itemIndex} className="table-col">
                                        {firstItem} {secondItem}
                                    </td>
                                )
                            })
                        }
                    </tr>
                )
            })
        )
    }

    const TableColumnsLenghts = () => {
        return (
            <tr className="table-row">
                {
                    props.data.map((value, index) => {
                        return (
                            <td key={index} className="table-col">
                                { props.label } {value.length}
                            </td>
                        )
                    })
                }
            </tr>
        )
    }

    return (
        props.data !== undefined
        && <tbody id="table-body">
            <TableData />
            <TableColumnsLenghts />
        </tbody>
    )

}

export default TableBody