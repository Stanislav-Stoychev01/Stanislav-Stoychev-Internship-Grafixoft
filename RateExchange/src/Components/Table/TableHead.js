function TableHead(props) {

    return (
        props.columnNames !== undefined &&
        <thead className = "table-head">
            <tr>
            {
                props.columnNames.map((item, index) => {
                    return (
                        <th key = { index }>{ item }</th>
                    )
                })  
            }  
            </tr>
        </thead >    
    )
}

export default TableHead