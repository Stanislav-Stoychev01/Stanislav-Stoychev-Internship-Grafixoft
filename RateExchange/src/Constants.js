export const currencyList = ['USD', 'EUR', 'AUD', 'CAD', 'CHF', 'NZD', 'BGN'];

export const defaultCurrency = 'USD';

export const maximumDifference = 0.5;

export const value = 1;

export const date = 'date';
export const latest = 'latest';

export const updateButtonName = "Update Currency";
export const inputLabel = "Input Currency Name:";
export const exchangeRateLabel = "Exchange Rates from";
export const longestSequenceMessage = "The longest sequence has length of ";
export const LoadingData = 'The data is currenly loading';
export const dropdownLabel = "Currency: ";
export const tableColumnsLenghtsLabel = "Count: ";
export const updateDateButtonText = "Update Date";

export const tableColumnNames = ['Conversion Rates below 1', 
'Conversion Rates between 1 and 1,5', 'Conversion Rates above 1,5'];

