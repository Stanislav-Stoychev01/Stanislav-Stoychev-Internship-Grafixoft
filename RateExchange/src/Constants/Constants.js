export const CurrencyList = ['USD', 'EUR', 'AUD', 'CAD', 'CHF', 'NZD', 'BGN']
export const DefaultCurrency = 'USD'
