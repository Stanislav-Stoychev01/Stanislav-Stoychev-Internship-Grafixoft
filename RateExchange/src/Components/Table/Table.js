import { useEffect, useState } from 'react';
import * as Constants from "../../Constants"
import * as CurrencyFilters from '../../Utils/CurrencyFilters';
import TableHead from "./TableHead";
import TableBody from './TableBody';
import "../../CSS/Table.css"

function Table(props)
{
    const [items, setItems] = useState();
    
    useEffect(() => {    
        if (props.data !== null) {
            const filteredResponse = CurrencyFilters.applyFilters(props.data);
            setItems(filteredResponse);
        } 
    }, [props.data]);

    if (items !== undefined) {
        return (
            <div> 
                <div className="current-date">
                <label className="exchange-date-label">
                    { props.label } { props.date }
                </label>
            </div>
            <div id ="table-component">
                <table id="table">
                    <TableHead 
                        columnNames = { Constants.tableColumnNames }/>
                    <TableBody
                        data = { items }
                        label={Constants.tableColumnsLenghtsLabel}/>
                </table>
            </div>
        </div>
        );
    }
    else{   
        return (
            <label className='label-loading'>
                { props.loading }
            </label>
        );
    }
}

export default Table;
