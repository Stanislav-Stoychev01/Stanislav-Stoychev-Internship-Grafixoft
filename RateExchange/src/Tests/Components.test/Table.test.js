import Table from "../../Components/Table/Table";
import * as Constants from "../../Constants";
import Enzyme from 'enzyme';
import toJson from "enzyme-to-json";
import { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-17-updated';

Enzyme.configure({ adapter: new Adapter() });

it('renders table with label when all props are provided', () => {

    const data = {
        'cad': '1.759853',
        'aud': '0.014107',
        'bgn': '2.618904',
        'eur': '3.673103',
        'afn': '85.586079'
    };

    const date = '2022-08-22';
    const currencyName = 'usd';

    const props = {
        label: Constants.exchangeRateLabel,
        data: data,
        currency: currencyName,
        date: date,
        loading: Constants.LoadingData,
    }

    const wrapper = mount(<Table {...props}/>);

    const jsonWrapper = toJson(wrapper);

    expect(jsonWrapper).toMatchSnapshot();
})

it("renders table when data is still loading", () => {
    let currencyName = 'usd';

    const response = null;

    const props = {
        label: Constants.exchangeRateLabel,
        data: currencyName,
        currency: currencyName,
        date: null,
        loading: Constants.LoadingData,
    };

    const wrapper = mount(<Table {...props} />);

    const jsonWrapper = toJson(wrapper);

    expect(jsonWrapper).toMatchSnapshot();
})