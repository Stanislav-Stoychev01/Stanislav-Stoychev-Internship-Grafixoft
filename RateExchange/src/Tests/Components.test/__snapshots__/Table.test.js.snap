// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders table when data is still loading 1`] = `
<Table
  currency="usd"
  data="usd"
  date={null}
  label="Exchange Rates from"
  loading="The data is currenly loading"
>
  <LoadingData />
  <RenderTable>
    <div
      className="testClass"
    >
      <div
        className="current-date"
      >
        <label
          className="exchange-date-label"
        >
          Exchange Rates from
           
        </label>
      </div>
      <div
        id="table-component"
      >
        <table
          id="table"
        >
          <TableHead
            columnNames={
              Array [
                "Conversion Rates below 1",
                "Conversion Rates between 1 and 1,5",
                "Conversion Rates above 1,5",
              ]
            }
          >
            <thead
              className="table-head"
            >
              <tr>
                <th
                  key="0"
                >
                  Conversion Rates below 1
                </th>
                <th
                  key="1"
                >
                  Conversion Rates between 1 and 1,5
                </th>
                <th
                  key="2"
                >
                  Conversion Rates above 1,5
                </th>
              </tr>
            </thead>
          </TableHead>
          <TableBody
            countField="Count: "
            data={
              Array [
                Array [],
                Array [],
                Array [],
              ]
            }
          >
            <tbody
              id="table-body"
            >
              <TableData />
              <TableColumnsLenghts>
                <tr
                  className="table-row"
                >
                  <td
                    className="table-col"
                    key="0"
                  >
                    Count: 
                     
                    0
                  </td>
                  <td
                    className="table-col"
                    key="1"
                  >
                    Count: 
                     
                    0
                  </td>
                  <td
                    className="table-col"
                    key="2"
                  >
                    Count: 
                     
                    0
                  </td>
                </tr>
              </TableColumnsLenghts>
            </tbody>
          </TableBody>
        </table>
      </div>
    </div>
  </RenderTable>
</Table>
`;

exports[`renders table with label when all props are provided 1`] = `
<Table
  currency="usd"
  data={
    Object {
      "afn": "85.586079",
      "aud": "0.014107",
      "bgn": "2.618904",
      "cad": "1.759853",
      "eur": "3.673103",
    }
  }
  date="2022-08-22"
  label="Exchange Rates from"
  loading="The data is currenly loading"
>
  <LoadingData />
  <RenderTable>
    <div
      className="testClass"
    >
      <div
        className="current-date"
      >
        <label
          className="exchange-date-label"
        >
          Exchange Rates from
           
          2022-08-22
        </label>
      </div>
      <div
        id="table-component"
      >
        <table
          id="table"
        >
          <TableHead
            columnNames={
              Array [
                "Conversion Rates below 1",
                "Conversion Rates between 1 and 1,5",
                "Conversion Rates above 1,5",
              ]
            }
          >
            <thead
              className="table-head"
            >
              <tr>
                <th
                  key="0"
                >
                  Conversion Rates below 1
                </th>
                <th
                  key="1"
                >
                  Conversion Rates between 1 and 1,5
                </th>
                <th
                  key="2"
                >
                  Conversion Rates above 1,5
                </th>
              </tr>
            </thead>
          </TableHead>
          <TableBody
            countField="Count: "
            data={
              Array [
                Array [
                  Array [
                    "aud",
                    "0.014107",
                  ],
                ],
                Array [],
                Array [
                  Array [
                    "cad",
                    "1.759853",
                  ],
                  Array [
                    "bgn",
                    "2.618904",
                  ],
                  Array [
                    "eur",
                    "3.673103",
                  ],
                ],
              ]
            }
          >
            <tbody
              id="table-body"
            >
              <TableData>
                <tr
                  className="table-row"
                  key="0"
                >
                  <td
                    className="table-col"
                    key="0"
                  >
                    aud
                     
                    0.014107
                  </td>
                  <td
                    className="table-col"
                    key="1"
                  >
                     
                  </td>
                  <td
                    className="table-col"
                    key="2"
                  >
                    cad
                     
                    1.759853
                  </td>
                </tr>
                <tr
                  className="table-row"
                  key="1"
                >
                  <td
                    className="table-col"
                    key="0"
                  >
                     
                  </td>
                  <td
                    className="table-col"
                    key="1"
                  >
                     
                  </td>
                  <td
                    className="table-col"
                    key="2"
                  >
                    bgn
                     
                    2.618904
                  </td>
                </tr>
                <tr
                  className="table-row"
                  key="2"
                >
                  <td
                    className="table-col"
                    key="0"
                  >
                     
                  </td>
                  <td
                    className="table-col"
                    key="1"
                  >
                     
                  </td>
                  <td
                    className="table-col"
                    key="2"
                  >
                    eur
                     
                    3.673103
                  </td>
                </tr>
              </TableData>
              <TableColumnsLenghts>
                <tr
                  className="table-row"
                >
                  <td
                    className="table-col"
                    key="0"
                  >
                    Count: 
                     
                    1
                  </td>
                  <td
                    className="table-col"
                    key="1"
                  >
                    Count: 
                     
                    0
                  </td>
                  <td
                    className="table-col"
                    key="2"
                  >
                    Count: 
                     
                    3
                  </td>
                </tr>
              </TableColumnsLenghts>
            </tbody>
          </TableBody>
        </table>
      </div>
    </div>
  </RenderTable>
</Table>
`;
